pkg/deploy/graph/analysis/dc.go

```diff
@@ -27,43 +29,54 @@ const (
 func FindDeploymentConfigTriggerErrors(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 	markers := []osgraph.Marker{}
 
-dc:
 	for _, uncastDcNode := range g.NodesByKind(deploygraph.DeploymentConfigNodeKind) {
-		for _, uncastIstNode := range g.PredecessorNodesByEdgeKind(uncastDcNode, deployedges.TriggersDeploymentEdgeKind) {
-			if istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode); !istNode.Found() {
-				dcNode := uncastDcNode.(*deploygraph.DeploymentConfigNode)
-
-				// The image stream for the tag of interest does not exist.
-				// TODO: Suggest `oc create imagestream` once we have that.
-				if isNode, exists := doesImageStreamExist(g, uncastIstNode); !exists {
-					markers = append(markers, osgraph.Marker{
-						Node:         uncastDcNode,
-						RelatedNodes: []graph.Node{uncastIstNode, isNode},
-
-						Severity: osgraph.ErrorSeverity,
-						Key:      MissingImageStreamErr,
-						Message: fmt.Sprintf("The image trigger for %s will have no effect because %s does not exist.",
-							f.ResourceName(dcNode), f.ResourceName(isNode)),
-					})
-					continue dc
+		dcNode := uncastDcNode.(*deploygraph.DeploymentConfigNode)
+		marker := ictMarker(g, f, dcNode)
+		if marker != nil {
+			markers = append(markers, *marker)
+		}
+	}
+
+	return markers
+}
+
+func ictMarker(g osgraph.Graph, f osgraph.Namer, dcNode *deploygraph.DeploymentConfigNode) *osgraph.Marker {
+	for _, uncastIstNode := range g.PredecessorNodesByEdgeKind(dcNode, deployedges.TriggersDeploymentEdgeKind) {
+		if istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode); !istNode.Found() {
+			// The image stream for the tag of interest does not exist.
+			// TODO: Suggest `oc create imagestream` once we have that.
+			if isNode, exists := doesImageStreamExist(g, uncastIstNode); !exists {
+				return &osgraph.Marker{
+					Node:         dcNode,
+					RelatedNodes: []graph.Node{uncastIstNode, isNode},
+
+					Severity: osgraph.ErrorSeverity,
+					Key:      MissingImageStreamErr,
+					Message: fmt.Sprintf("The image trigger for %s will have no effect because %s does not exist.",
+						f.ResourceName(dcNode), f.ResourceName(isNode)),
 				}
+			}
+
+			if bcNode := buildedges.BuildConfigForTag(g, istNode); bcNode != nil && !istNode.Found() {
```

Comment:
Can we be sure that this will work as necessary with regards to the one-to-many relationship of ISTag::BC ? We want to skip the warning marker if *any* BC pointing to the ISTag has an unfinished build, right?